happi/cli.py

```diff
@@ -132,8 +132,16 @@ def search(
             continue
 
         elif is_number(value):
-            logger.debug('Changed %s to float', value)
-            # value = str(float(value))
+            if float(value) == int(float(value)):
+                # value is an int, allow the float version (optional .0)
+                logger.debug(f'looking for int value: {value}')
+                value = f'^{int(float(value))}(\\.0+$)?$'
+
+                # don't translate from glob
+                client_args[criteria] = value
+                continue
+            else:
+                value = str(float(value))
```

Comment:
Couldn't a floating point value be somewhat problematic for regex mode? The above comment and continue gave me pause to think about it a bit:

* Say we passed in "3.14" we might match "3514" or "3x14" just the same. `.` would be interpreted as a regex anychar, wouldn't it?
* Does float-based value searching also work as a full match or a partial match? That is, if `12345.678` is in the database and I search for `5.678` what happens? (lazy question...)

Reply:
These are good points.  
- I think you're right, ints will always get an escaped `.`, but floats will get passed through as is.  Is this a problem though?  We could argue if someone is searching in regex mode they should be escaping their `.` if they need to?
- The float-based value searching seems to not work with partial matches, though I didn't do that intentionally.

Reply:
> We could argue if someone is searching in regex mode they should be escaping their . if they need to?

If someone is doing a regex search with a thing that looks like a floating point value... is it a floating point value or a regex?

> Is this a problem though? 

Likely not - just trying to dig into the implications of the changes in my usual excessive way 😁

Reply:
I think this is an annoyance with using regex in general to search for floating point numbers rather than a unique problem to happi

Float-based values not matching almost seems like a feature rather than a bug...

Reply:
That's a good point about the processing of regex, Ken

Reply:
> Likely not - just trying to dig into the implications of the changes in my usual excessive way 😁

I don't think it's excessive, it's just thorough 😄 

>Float-based values not partial matching almost seems like a feature rather than a bug...

I wish I could claim credit for that haha